parser/mpFuncCommon.cpp

```diff
@@ -856,6 +875,55 @@ MUP_NAMESPACE_START
     return new FunTimeDiff(*this);
   }
 
+  //------------------------------------------------------------------------------
+  //
+  // class FunCurrentTime
+  //
+  //------------------------------------------------------------------------------
+
+  FunCurrentTime::FunCurrentTime()
+    :ICallback(cmFUNC, _T("current_time"), -1)
+  {}
+
+  void FunCurrentTime::Eval(ptr_val_type &ret, const ptr_val_type *a_pArg, int a_iArgc)
+  {
+    int gmt_offset = 0;
+    if (a_iArgc > 1) {
+      throw ParserError(ErrorContext(ecTOO_MANY_PARAMS, GetExprPos(), GetIdent()));
+    } else if (a_iArgc == 1) {
+      switch(a_pArg[0]->GetType())
+      {
+      case 'i': gmt_offset = a_pArg[0]->GetInteger();   break;
+      default:
+        {
+          ErrorContext err;
+          err.Errc = ecTYPE_CONFLICT_FUN;
+          err.Arg = 1;
+          err.Type1 = a_pArg[0]->GetType();
+          err.Type2 = 'i';
+          throw ParserError(err);
+        }
+      }
+    }
+
+    std::time_t t = std::time(0);
+    std::tm now = *std::localtime(&t);
+
+    *ret = format_time(now, gmt_offset);
+  }
+
+  ////------------------------------------------------------------------------------
+  const char_type* FunCurrentTime::GetDesc() const
+  {
+    return _T("current_time() - Returns the current time in the HH:MM:SS format.");

```

Comment:
```suggestion
    return _T("current_time("+1") - Returns the current time in the HH:MM:SS format.");
```

Reply:
oh, i thought it best to leave it as an integer param, is that a problem? so it would be current_time(1) not current_time("+1"). otherwise it would need a parser for the offset

Reply:
Ahhh! Its fine as you did! Is missing tests for that, right?

Reply:
yeah! i was going to ping you when the pr was complete hehe. it is now ready